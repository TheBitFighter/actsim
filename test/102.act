deftype str (int<4> x)
{
   methods {
     macro plus (int<2> y) : int<4>
     {
       self := x + y
     }
   }
}

defproc test()
{
   str s;
   chp {
      s.x := 3;
      log ("got ", s.plus(2))
   }
}

test t;
