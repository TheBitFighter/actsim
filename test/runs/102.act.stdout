[                  10] <>  got 5
